draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -162,61 +199,92 @@ The following terms are used throughout this document:
-- Nominal Diffie-Hellman Group {{group}}.
+- Key Encapsulation Mechanism {{kems}}
+- Extendable Output Function (XOF) {{xof}}
+- Key Derivation Function (KDF) {{kdf}}
 
-These dependencies are defined in the following subsections.
+In the remainder of this section, we describe functional aspects of these
+mechanisms.  The security properties we require in order for the resulting
+hybrid KEM to be secure are discussed in {{security-properties}}.
 
 ## Key encapsulation mechanisms {#kems}
 
-Key encapsulation mechanisms (KEMs) are cryptographic schemes that consist of
-four algorithms:
-
-- `KeyGen() -> (pk, sk)`: A probabilistic key generation algorithm, which
-  generates a public encapsulation key `pk` and a secret decapsulation key
-  `sk`, each of which are byte strings.
-- `DeriveKey(seed) -> (pk, sk)`: A deterministic algorithm, which takes as
-  input a seed `seed` and generates a public encapsulation key `pk` and a
-  secret decapsulation key `sk`, each of which are byte strings.
-- `Encaps(pk) -> (ct, shared_secret)`: A probabilistic encapsulation
-  algorithm, which takes as input a public encapsulation key `pk` and outputs
-  a ciphertext `ct` and shared secret `shared_secret`.
-- `Decaps(sk, ct) -> shared_secret`: A decapsulation algorithm, which takes
-  as input a secret decapsulation key `sk` and ciphertext `ct` and outputs a
-  shared secret `shared_secret`.
-
-KEMs can also provide a deterministic version of `Encaps`, denoted
-`EncapsDerand`, with the following signature:
-
-- `EncapsDerand(pk, randomness) -> (ct, shared_secret)`: A deterministic
+~~~ aasvg
+     +-----------------+
+     | GenerateKeyPair |
+     |        or       |
+     |  DeriveKeyPair  |
+     +--------+--------+
+              |
+    +---------+----------+
+    |                    |
+    V                    V
+
+    ek                  dk
+
+    |                    |
+    |                    |
+    V                    V
++--------+    ct    +--------+
+| Encaps |--------->| Decaps |
++--------+          +--------+
+    |                    |
+    |                    |
+    V                    V
+
+    ss        ==        ss
+~~~
+
+A Key Encapsulation Mechanism (KEMs) comprises the following algorithms:
+
+- `GenerateKeyPair() -> (ek, dk)`: A randomized algorithm that generates a
+  public encapsulation key `ek` and a secret decapsulation key `dk`, each of
+  which are byte strings.
+- `DeriveKeyPair(seed) -> (ek, dk)`: A deterministic algorithm that takes as
+  input a seed `seed` and generates a public encapsulation key `ek` and a secret
+  decapsulation key `dk`, each of which are byte strings.
+- `Encaps(ek) -> (ct, ss)`: A probabilistic encapsulation
+  algorithm, which takes as input a public encapsulation key `ek` and outputs
+  a ciphertext `ct` and shared secret `ss`.
+- `Decaps(dk, ct) -> ss`: A decapsulation algorithm, which takes
+  as input a secret decapsulation key `dk` and ciphertext `ct` and outputs a
+  shared secret `ss`.
```

Comment:
'Yes but', in the literature there is just keygen, so idk if we want to add a word or two to like, make this more 'applied', or applicable just in this document, or something